tests/dialects/test_duckdb.py

```diff
@@ -753,13 +753,15 @@ def test_duckdb(self):
             },
         )
 
-        with self.assertRaises(UnsupportedError):
-            transpile(
-                "SELECT REGEXP_EXTRACT(a, 'pattern', 1) from table",
-                read="bigquery",
-                write="duckdb",
-                unsupported_level=ErrorLevel.IMMEDIATE,
-            )
+        self.validate_all(
```

Comment:
We should add an identity test in `duckdb.py`, making sure that the option stored in `parameters` is not dropped during roundtrip.

I also see that dialects like [Snowflake](https://docs.snowflake.com/en/sql-reference/functions/regexp_substr) support these options/parameters, would also be good to add transpilation tests between them.

Reply:
Sounds good on first one, I see how to do that .

On second, do I need to add tests to test_snkwflake.py for that? I don't really understand the first arg to this assert function, is it supposed to be in the dialect of `self`, or is it the dialect that the internal RegexpExtract uses?

Reply:
Right, takes some time getting used to these :)

- Identity tests: These are tests that validate the roundtrip (i.e parsed DuckDB -> generated DuckDB). The API to do that is `self.validate_identity(query)`, or if the roundtrip slightly alters the result we can do `self.validate_identity(query_before, query_after)`
- Transpilation tests: Most usually done through `self.validate_all` in a similar fashion.

In both these cases `self` refers to the dialect in question, yes. So if you wanted to test `REGEXP_EXTRACT` between DuckDB and Snowflake, you could do:

```
# test_duckdb.py

self.validate_all(
 "REGEXP_EXTRACT(arg1, arg2, arg3)",
  read = {
   "snowflake": "REGEXP_SUBSTR(arg1, arg2, arg3, arg4)",
  },
  write = {
   "snowflake": "REGEXP_SUBSTR(arg1, arg2, arg3)"
  },
)
```

So, for any `validate_all(query, read_dict, write_dict)` call we expect that:
- All the queries in the `read` dialects will be parsed in their respective dialects and will be transpiled to the `self` dialect (DuckDB in this case) resulting in `query`
- Parsing `query` in the `self` dialect and transpiling that to the `write` dialects will result in the respective queries

This way, we can test both directions of a transpilation path (e.g. Snowflake -> Duckdb and Duckdb -> Snowflake) in a single call

It'll help if you invest some time to read the definitions of these functions, they're only a few in number.

Reply:
OK, I think I got many of the variations. Let me know if you see some more that we should catch